### ISSUE TITLE
Larger touch area on checkbox and radio inputs

### ISSUE THREAD

**Issue by Febakke** [digdir/designsystem#4003]
## Should we increase the touch area of checkboxes and radio inputs?
As of today our inputs have a 24x24 toucharea. The question is if we should create an invisible area that is counted as the touch area.

**We need to look at**
- WCAG, what are the revelant requirements we need to look at
- Design: How would this work in practice? Do we need more space between elements? Avoid overlap
- Technical, what are up and downsides of different technical solutions if we decide 

### WCAG
There are at this moment not any WCAG requirements that are required by Norwegian law. But there are some guidelines. 

[Target size (enhanced)](https://www.w3.org/WAI/WCAG22/Understanding/target-size-enhanced.html)
This requires that all touch areas are at least 44x44. 

[Target size (minimum) ](https://www.w3.org/WAI/WCAG22/Understanding/target-size-minimum.html)
This is newer requirement that I think will be a part of the norwegian requirements at a future date. This is set as 24x24

### Technical

**Comment by Barsnes** [comment]
Some quick notes:
`<label>` is also touchable, and could maybe be considered in the touch area? 
We can't use pseudo elements (`::before`, `::after`) since these are not supported in `<input>`

**Comment by oddvernes** [comment]
Perhaps hard to avoid needing a wrapper. Here is how we made checkbox in equinor with 48x48 physical area, 24x24 visible area and 40x40 hover circle area https://storybook.eds.equinor.com/?path=/docs/inputs-selection-controls-checkbox--docs  (inpect element to get the idea)

**Comment by Barsnes** [comment]
> Perhaps hard to avoid needing a wrapper. Here is how we made checkbox in equinor with 48x48 physical area, 24x24 visible area and 40x40 hover circle area https://storybook.eds.equinor.com/?path=/docs/inputs-selection-controls-checkbox--docs (inpect element to get the idea)

Thanks. 
Ideally we would like to solve this without wrappers, since using DS without React will become a bit more cumbersome.
A good documentation _could_ solve this, but at the time of writing we don't have a specific HTML/CSS doc.

Changing the DOM structure would also be considered a breaking change, unless we also support the old structure

**Comment by oddvernes** [comment]
Yes agreed, no changes to dom should be done on inputs

**Comment by mrosvik** [comment]
We need to make a decision here.

**Comment by frankdahle** [comment]
I'm late to the party here, but anyway, I would argue that this is best practice and what we should implement our components like this, both regular radio/checkbox and panels. With this I recommend that the following also apply:
- The height should follow WCAG 2.5.5 target size 44px https://www.w3.org/WAI/WCAG21/Understanding/target-size.html. 
- The width should be the same for all radios/checks within a group, regardless of the lenght of the label.
- All states should cover the whole component (hover, focus, disabled).

<img width="465" height="318" alt="Image" src="https://github.com/user-attachments/assets/db9ae3df-abd6-430c-894b-1e6439a6d37e" />

**Comment by eirikbacker** [comment]
The full content of the `label` is read outloud by screen readers (and placed in the Screen Reader rotor). If we place _a lot_ of content in the `label`, this will cause a quite bad experience accessibility wise.
I suggest we instead use the `data-clickdelegatefor=""` attribute as we are implement in #4335, and let the `label` only surround the main text/title. This means we can re-use the card-markup/component, we get a nice CSS-hover-logic, and the click surface is expanded to the whole card/container.

**Comment by Febakke** [comment]
@eirikbacker 
Are you thinking about the radioPanel variant of radiobutton? or could `data-clickdelegatefor=""` also be used to expand the clickable area of a standard radio/checkbox?

**Comment by eirikbacker** [comment]
> [@eirikbacker](https://github.com/eirikbacker) Are you thinking about the radioPanel variant of radiobutton? or could `data-clickdelegatefor=""` also be used to expand the clickable area of a standard radio/checkbox?

Unsure if I understand the question, but `data-clickdelegatefor=""` can be used to expand the clickable area of all radio/checkbox/button/links - regardless of design or component. This is a generic solution allowing you to "expand" the clickable area of a interactive element, to any parent container :)

**Comment by Barsnes** [comment]
> > [@eirikbacker](https://github.com/eirikbacker) Are you thinking about the radioPanel variant of radiobutton? or could `data-clickdelegatefor=""` also be used to expand the clickable area of a standard radio/checkbox?
> 
> Unsure if I understand the question, but `data-clickdelegatefor=""` can be used to expand the clickable area of all radio/checkbox/button/links - regardless of design or component. This is a generic solution allowing you to "expand" the clickable area of a interactive element, to any parent container :)

What @frankdahle is talking about here is any generic checkbox.
He point out that both the input and label should be clikable, which it is today.
In a meeting yesterday he also mentioned that all checkboxes should have the same clickarea, and that it should be larger then just input + label. This is easily solved by setting the `<label>` width to 100%. I don't think we can do this safely, but it is quite simple to solve for consumers in places where this is applicable.

Expect for Franks second point, we have everything today.

**Comment by Barsnes** [comment]
And this issue, as far as I understand, is not related to our `*Panel` components we have planned.

**Comment by eirikbacker** [comment]
Thanks for clarifying @Barsnes ☺️ 
In regards of `label` with `width: 100%` - we had this in our system, but received pushback as users found it confusing that the clickable area was _much_ wider than what they saw in the interface, and could even by accident click something.

I very much agree the click area should be the label _and_ input, but I do not see the need for them to expand much as we are well inside the requirements for accessibility ([24x24px according to WCAG 2.5.8](https://www.w3.org/TR/WCAG22/#target-size-minimum), and our radio/checkboxes are `27px` high and `44px` wide even with a single character label) from my understanding, or is there something I might have overlooked @frankdahle ? ☺️

**Comment by frankdahle** [comment]
@eirikbacker , those users who pushed back, were they within the accessibility target group? I've had lots of positive respons during the years, from user testing with both capable and less capable users. So there you go :) Remember also that WCAG is the absolute minimum requirements. 27x27px is a very small area. On the other hand, it doesn't have to be 44px to improve ux significantly. I did some more critical searching to be sure, and I still find clear support for larger click areas. But of course, if a certain design makes users accidentily click on things without meaning to, something is wrong with the desing in those cases. let's have a look at some more examples later, and discuss it more.

**Comment by mrosvik** [comment]
We should also make a blog post about our decision.

**Comment by Febakke** [comment]
@frankdahle 
It seems the suggested solution for this caused other issues when combining radio/checkboxes with other components inside a `fieldset` this would break products that have already implemented this. 

I would argue that we do not implement a larger touch area for our default radio/checkbox components as there are no way for us to implement this without causing issues for our users. We are not breaking Target size AA, and with the vertical spacing we are going beyond the minimum target. If there is a concern that the target size still is to small, we need research before we do changes to the component that impact our users. 

As an alternative we are making the radio/checkbox panel variant that will go beyond AAA that we can recommend our user to use.

**Comment by mrosvik** [comment]
Will be partly solved by #526. 
+ Our components scale with data-size, so those who need larger touch areas can solve this themselves.
